### PR TITLE
Using the new cred API for multiple auxiliary modules

### PR DESCRIPTION
**Pull request by wchen-r7**
This is what I have for today, feel free to review/merge anytime. There will be more.

To test this:
- [x] Start msfconsole
- [x] Do: `loadpath [path to msf/test/modules/]`
- [x] Do: `use auxiliary/test/report_auth_info`
- [x] Do: `run`
- [x] You should not see any backtrace

### CODE REVIEW

**Comment by void-in** [comment]
+1 for having the modules in a single PR. It is easy to review. Just a minor issue @wchen-r7. In the module modules/auxiliary/scanner/http/drupal_views_user_enum.rb, you are not calling the report_cred method and it is still using report_auth_info on line 132. Just need to replace it with report_cred.

**Comment by wchen-r7** [comment]
Ah, nice catch! Thanks, I'll fix it.

File: modules/auxiliary/scanner/http/cisco_ssl_vpn.rb [hunk 1]
```diff
@@ -197,18 +224,7 @@ def do_login(user, pass, group)
 
         do_logout(resp.get_cookies)
 
-        report_hash = {
-          :host   => rhost,
-          :port   => rport,
-          :sname  => 'Cisco SSL VPN',
-          :user   => user,
-          :pass   => pass,
-          :group => group,
```
Comment (jvazquez-r7) [inline modules/auxiliary/scanner/http/cisco_ssl_vpn.rb]: I feel like the group is an important piece of information in order to login. I don't know what is the best way, but I think would worth to find a way of report it with `report_cred`

File: modules/auxiliary/scanner/http/splunk_web_login.rb [hunk 2]
```diff
@@ -165,15 +190,8 @@ def do_login(user, pass)
...
-      report_hash = {
-        :host   => datastore['RHOST'],
-        :port   => datastore['RPORT'],
-        :sname  => 'splunk-web',
-        :user   => user,
-        :pass   => pass,
-        :active => true,
-        :type => 'password'}
 
       report_auth_info(report_hash)
```
Comment (jvazquez-r7) [inline modules/auxiliary/scanner/http/splunk_web_login.rb]: Should it be deleted?

File: modules/auxiliary/scanner/http/cisco_ssl_vpn.rb (hunk 1 above)
Comment (wchen-r7) [inline modules/auxiliary/scanner/http/cisco_ssl_vpn.rb]: report_note?

File: modules/auxiliary/scanner/http/splunk_web_login.rb (hunk 2 above)
Comment (wchen-r7) [inline modules/auxiliary/scanner/http/splunk_web_login.rb]: eh, yup

File: modules/auxiliary/scanner/http/drupal_views_user_enum.rb [hunk 3]
```diff
@@ -63,6 +63,30 @@ def check_host(ip)
...
+    credential_data = {
+      origin_type: :service,
+      module_fullname: fullname,
+      username: opts[:user]
+    }.merge(service_data)
+
+    login_data = {
+      last_attempted_at: DateTime.now,
+      core: create_credential(credential_data),
+      status: Metasploit::Model::Login::Status::SUCCESSFUL,
```
Comment (jvazquez-r7) [inline modules/auxiliary/scanner/http/drupal_views_user_enum.rb]: It is an user enumeration module. So there is no password, there is just an UNTRIED username.

File: modules/auxiliary/scanner/http/cisco_ssl_vpn.rb (hunk 1 above)
Comment (jvazquez-r7) [inline modules/auxiliary/scanner/http/cisco_ssl_vpn.rb]: yup, that could work :)

**Comment by jvazquez-r7** [comment]
Other the minor comments are fixed it should be good enough to go!:

```
msf auxiliary(report_auth_info) > run

[*] Trying: #test_novell_mdm_creds
[+] That didn't blow up. Good!

[*] Trying: #test_joomla_bruteforce_login
[+] That didn't blow up. Good!

[*] Trying: #test_infovista_enum
[+] That didn't blow up. Good!

[*] Trying: #test_etherpad_duo_login
[+] That didn't blow up. Good!

[*] Trying: #test_ektron_cms400net
[+] That didn't blow up. Good!

[*] Trying: #test_drupal_views_user_enum
[+] That didn't blow up. Good!

[*] Trying: #test_dolibarr_login
[+] That didn't blow up. Good!

[*] Trying: #test_dlink_dir_session_cgi_http_login
[+] That didn't blow up. Good!

[*] Trying: #test_dlink_dir_615h_http_login
[+] That didn't blow up. Good!

[*] Trying: #test_dlink_dir_300_615_http_login
[+] That didn't blow up. Good!

[*] Trying: #test_cisco_ssl_vpn
[+] That didn't blow up. Good!

[*] Trying: #test_cisco_ironport_enum
[+] That didn't blow up. Good!

[*] Trying: #test_cisco_asa_asdm
[+] That didn't blow up. Good!

[*] Trying: #test_axis_local_file_include
[+] That didn't blow up. Good!

[*] Trying: #test_splunk_web_login
[+] That didn't blow up. Good!

[*] Trying: #test_cctv_dvr_login
[+] That didn't blow up. Good!

[*] Trying: #test_http_vcms_login
[+] That didn't blow up. Good!

[*] Trying: #test_telnet_ruggedcom
[+] That didn't blow up. Good!

[*] Trying: #test_vmware_http_login
[+] That didn't blow up. Good!

[*] Trying: #test_asterisk_login
[+] That didn't blow up. Good!

[*] Auxiliary module execution completed
msf auxiliary(report_auth_info) > creds
Credentials
===========

host            service                            public    private   realm  private_type
----            -------                            ------    -------   -----  ------------
192.168.12.0    80/tcp (novellmdm)                 username  password         Password
192.168.12.1    80/tcp (https)                     username  password         Password
192.168.12.2    80/tcp (InfoVista VistaPortal)     username  password         Password
192.168.12.3    80/tcp (https)                     username  password         Password
192.168.12.4    80/tcp (https)                     username  password         Password
192.168.12.5    80/tcp (https)                     username
192.168.12.6    80/tcp (https)                     username  password         Password
192.168.12.7    80/tcp (https)                     username  password         Password
192.168.12.8    80/tcp (https)                     username  password         Password
192.168.12.9    80/tcp (https)                     username  password         Password
192.168.12.10   80/tcp (Cisco SSL VPN)             username  password         Password
192.168.12.11   80/tcp (Cisco IronPort Appliance)  username  password         Password
192.168.12.12   80/tcp (Cisco ASA ASDM)            username  password         Password
192.168.12.13   80/tcp (https)                     username  password         Password
192.168.12.14   80/tcp (splunk-web)                username  password         Password
192.168.12.15   80/tcp (cctv_dvr)                  username  password         Password
192.168.12.16   80/tcp (http)                      username  password         Password
192.168.12.17   80/tcp (telnet)                    factory   password         Password
192.168.12.18   80/tcp (vmware)                    username  password         Password
192.168.12.19   80/tcp (asterisk_manager)          username  password         Password
```

File: modules/auxiliary/scanner/http/cisco_ssl_vpn.rb (hunk 1 above)
Comment (wchen-r7) [inline modules/auxiliary/scanner/http/cisco_ssl_vpn.rb]: Ok I'll do that.

**Comment by wchen-r7** [comment]
Ok give me like 5 - 10 minutes, updating the bugs.

**Comment by wchen-r7** [comment]
Ok I think I got them all. Thanks!

**Comment by jvazquez-r7** [comment]
Thanks @wchen-r7 landing!

**Pull request merged**